### PR TITLE
Add new package rlwrap v0.46.1

### PR DESCRIPTION
**Pull request by saranyareddy24**
Adding a new package named rlwrap to SPECS-EXTENDED.

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Change Log  

- Adding new package rlwrap
- 
###### Does this affect the toolchain?  
NO

###### Test Methodology
- local build
- Pipeline Id: [Unified BuddyBuild](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=388965&view=results)

Note -- On behalf of @saranyareddy24:
- Source uploaded by @mfrw
- Buddy Build triggered by @mfrw

### CODE REVIEW

**Comment by saranyareddy24** [comment]
> @saranyareddy24 please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree [company="Microsoft"]

**Comment by saranyareddy24** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Comment by saranyareddy24** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: SPECS-EXTENDED/rlwrap/rlwrap.signatures.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+ "Signatures": {
+  "rlwrap-0.46.1.tar.gz": "2711986a1248f6ac59e2aecf5586205835970040d300a42b4bf8014397e73e37"
```
Comment (mfrw) [inline SPECS-EXTENDED/rlwrap/rlwrap.signatures.json:3]: Looks like the sources are not uploaded to the blob store.
I will upload that in the meantime.

Comment (saranyareddy24) [reply SPECS-EXTENDED/rlwrap/rlwrap.signatures.json:3]: Thank you.

File: SPECS-EXTENDED/rlwrap/rlwrap.spec [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+%doc AUTHORS NEWS README.md
+%{_bindir}/rlwrap
+%{_mandir}/*/rlwrap.*
+%{_mandir}/man3/RlwrapFilter.*
+%{_datadir}/rlwrap
+
+
+%changelog
+* Thu Jun 22 2023 Saranya R <rsaranya@microsoft.com> - 0.46.1-1
+- Initial CBL-Mariner import from Fedora (license: MIT).
```
Comment (mfrw) [inline SPECS-EXTENDED/rlwrap/rlwrap.spec]: Can you please add what version of Fedora this spec was imported from ?
e.g: F38 has this version
```suggestion
- Initial CBL-Mariner import from Fedora 38 (license: MIT).
```

File: SPECS-EXTENDED/rlwrap/rlwrap.spec [hunk 3]
```diff
@@ -0,0 +1,73 @@
+Name:           rlwrap
+Version:        0.46.1
+Release:        1%{?dist}
+Summary:        Wrapper for GNU readline
+License:        MIT
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            https://github.com/hanslub42/rlwrap
+Source:         https://github.com/hanslub42/rlwrap/releases/download/%{version}/%{name}-%{version}.tar.gz
+
```
Comment (mfrw) [inline SPECS-EXTENDED/rlwrap/rlwrap.spec:10]: Would you please mind addressing the `linter` suggestions as well.

Comment (saranyareddy24) [reply SPECS-EXTENDED/rlwrap/rlwrap.spec]: done

**Review by mfrw — APPROVED** [review]
LGTM

**Pull request merged**